Raven.Database/Commercial/ValidateLicense.cs

```diff
@@ -125,10 +126,21 @@ private void ExecuteInternal(InMemoryRavenConfiguration config, bool firstTime =
                         }
 
                         var attributes = new Dictionary<string, string>(AlwaysOnAttributes, StringComparer.OrdinalIgnoreCase);
-                        foreach (var licenseAttribute in licenseValidator.LicenseAttributes)
+
+                        try
                         {
-                            attributes[licenseAttribute.Key] = licenseAttribute.Value;
+                            Monitor.Enter(licenseValidator.LicenseAttributesLock);
```

Comment:
I think that this should be outside `tryf` block?

Reply:
changed